homeassistant/components/recorder/migration.py

```diff
@@ -2222,7 +2289,18 @@ def __init__(self, schema_version: int, migration_changes: dict[str, int]) -> No
         self.schema_version = schema_version
         self.migration_changes = migration_changes
 
-    def do_migrate(self, instance: Recorder, session: Session) -> None:
+    def migrate_all(
+        self, instance: Recorder, session_maker: Callable[[], Session]
+    ) -> None:
+        """Migrate all data."""
+        with session_scope(session=session_maker()) as session:
+            if not self.needs_migrate(instance, session):
+                self.migration_done(instance, session)
+                return
+        while not self.migrate_data(instance):
```

Comment:
`migrate_data` is wrapped in `retryable_database_job` which never gives up. I'm not sure if this is OK here, or we should instead fail after a number of attempts?

Reply:
I think we should be using `database_job_retry_wrapper` which fails after 5 attempts